File: cmd/archigraph/testdata/discriminator_fixture/checklist.ts
// Fixture for issue #2670 — verify the production extract pipeline emits
// DISCRIMINATES_ON edges for discriminator-pattern comparisons in real files.

export function processChecklist(checklistType: number, role: string, status: string) {
  const isCat5 = checklistType === 2;
  const isAdmin = role === 'admin';
  const isActive = status === 'active';
  return isCat5 && isAdmin && isActive;
}

File: cmd/grafel/testdata/discriminator_fixture/checklist.py
# Fixture for issue #2670 — verify the production extract pipeline emits
# DISCRIMINATES_ON edges for discriminator-pattern comparisons in real files.


def process_checklist(checklist_type, role, status):
    is_cat5 = checklist_type == 2
    is_admin = role == "admin"
    is_active = status == "active"
    return is_cat5 and is_admin and is_active
